tools/common/Application.cs

```diff
@@ -524,10 +524,10 @@ public void RunRegistrar ()
 			var resolver = new PlatformResolver () {
 				FrameworkDirectory = Driver.GetPlatformFrameworkDirectory (this),
 				RootDirectory = Path.GetDirectoryName (RootAssembly),
-			};
 #if MMP
-			resolver.RecursiveSearchDirectories.AddRange (Driver.RecursiveSearchDirectories);
+				CommandLineAssemblies = RootAssemblies,
```

Comment:
This is what actually makes the MonoMacResolver actually resolve to assemblies given as arguments. I can add another list to be checked, but I think this is fine.

1. This code path is `--runregistrar` specific
2. The other bit of code which sets this is inside `Pack`, which is the _other_ hit when the action is not `--runregistrar`

https://github.com/xamarin/xamarin-macios/blob/3113c5d2b5cca730372177e9b1cee17062a959ff/tools/mmp/driver.cs#L513

Reply:
I agree that this is the correct fix at face value. And I'm totally cool with it going into master.

I've just been burned enough times with last minute behavior changes in release branches that i'm hesitant.

Reply:
Want me to split the PR into 2 commits? One which reverts recursive search dirs and one which implements this?

So we can backport the revert safely?

Reply:
That would be grand. ❤️

Reply:
Done.